kubernetes/utils/create_from_yaml.py

```diff
@@ -59,47 +51,74 @@ def create_from_yaml(
         directive will result in an error response and no further
         processing of the request.
         Valid values are: - All: all dry run stages will be processed
-    """
 
+    Raises:
+        FailToCreateError which holds list of `client.rest.ApiException`
+        instances for each object that failed to create.
+    """
     with open(path.abspath(yaml_file)) as f:
         yml_document_all = yaml.safe_load_all(f)
-        api_exceptions = []
-        # Load all documents from a single YAML file
+
+        failures = []
```

Comment:
the indentation level can be lowered here
one does not need the file descriptor anymore at this point as the yaml was already loaded

Reply:
This is incorrect. See my git commit log:

    ```
    with open('foo.txt') as f:
        y = yaml.safe_load_all(f)
    
    for i in y:
        print(i)
    
    \# raises ValueError: I/O operation on closed file.
    ```
    
    Hence, we indent the whole method body into the open block.
    
    with open('foo.txt') as f:
        y = yaml.safe_load_all(f)
        for i in y:
            print(i)

Reply:
oh it returns a generator, I forgot about that, nevermind then